tests/rust_syscall_tests/src/lib.rs

```diff
@@ -0,0 +1,516 @@
+    } else {
+        0usize
+    };
+
+    let arg2 = if let Some(buf) = buf {
+        buf.as_mut_ptr() as usize
+    } else {
+        0usize
+    };
+
+    unsafe {
+        asm!(
+        "syscall",
+        inlateout("rax") SYS_GETATT => rax,
+        in("rdi") arg0,
+        in("rsi") arg1,
+        inlateout("rdx") arg2 => rdx,
+        in("r10") arg3,
+        in("r8") 0,
+        in("r9") 0,
+        lateout("rcx") _, // clobbered
+        lateout("r11") _, // clobbered
+        );
+    }
+
+    if rax < 0 {
+        return Err(-rax as _);
+    }
+
+    let tech = TeeTech::try_from(rdx).map_err(|_| libc::EINVAL)?;
+
+    Ok((rax as _, tech))
+}
+
+pub fn getuid() -> Result<libc::uid_t> {
+    let ret = syscall(libc::SYS_getuid, Args::default()).0 as isize;
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn geteuid() -> Result<libc::uid_t> {
+    let ret = syscall(libc::SYS_geteuid, Args::default()).0 as isize;
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn getgid() -> Result<libc::uid_t> {
+    let ret = syscall(libc::SYS_getgid, Args::default()).0 as isize;
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn getegid() -> Result<libc::uid_t> {
+    let ret = syscall(libc::SYS_getegid, Args::default()).0 as isize;
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn uname(buf: *mut libc::utsname) -> Result<()> {
+    let ret = syscall(
+        libc::SYS_uname,
+        Args {
+            arg0: buf as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret == 0 {
+        Ok(())
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn socket(domain: i32, typ: i32, protocol: i32) -> Result<i32> {
+    let ret = syscall(
+        libc::SYS_socket,
+        Args {
+            arg0: domain as _,
+            arg1: typ as _,
+            arg2: protocol as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn bind(sockfd: i32, addr: *const libc::sockaddr, addrlen: libc::socklen_t) -> Result<i32> {
+    let ret = syscall(
+        libc::SYS_bind,
+        Args {
+            arg0: sockfd as _,
+            arg1: addr as _,
+            arg2: addrlen as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn listen(sockfd: i32, backlog: i32) -> Result<i32> {
+    let ret = syscall(
+        libc::SYS_listen,
+        Args {
+            arg0: sockfd as _,
+            arg1: backlog as _,
+            ..Default::default()
+        },
+    )
+    .0 as isize;
+
+    if ret >= 0 {
+        Ok(ret as _)
+    } else {
+        Err(-ret as i32)
+    }
+}
+
+pub fn accept(
```

Comment:
Are accept, accept4, connect, and recvfrom actually being used anywhere?

Reply:
true, it was in the old `lib.h` :) but does not hurt, if someone writes a test for it

Reply:
Should we have tests for them in this PR?

Reply:
no, not now.. this PR is just to replace the old C tests